mpi-proxy-split/README

```diff
@@ -1,94 +1,217 @@
+
+Before going into details, there are a few prerequisites:
+A.  We have only tested with MPICH, not Open MPI.  Open MPI doesn't
+    seem to support a statically linked MPI library.  (MANA uses a
+    static MPI library in its lower half, while the MPI application continues
+    to be linked dynamically  MANA's lower-half static MPI library can
+    be replaced by a standard dynamic MPI library, but this would requires
+    further work, and is not a current priority.)
+B.  The MPICH package supplied with CentOS 7 provides mpicc, which
+    accepts a '-static' flag.  But this is not sufficient for
+    actual use.  To configure MANA to work with MPICH and '-static', see:
+      MANA_ROOT_DIR/contrib/mpi-proxy-split/lower-half/README.mpich-static
+C.  Note MANA_ROOT_DIR/contrib/mpi-proxy-split/Makefile_configure.in
+    for additional customization (for example, not Cori, not MPICH).
+D.  OPTIONAL: Probably the 'configure' above suffices.  But if 'mpicc -show'
+    indicates a flag '-lxml2' anyway, then this will create a dependency on
+    libxml2.a.  Even if you have installed the package 'libxml2', then you
+    _must_ also install 'libxml2-dev' (libxml2-devel in CentOS) to get
+    'libxml2.a'.
+    (If you are missing libxml2.a and lack root privilege, try copying libxml2.a
+     from some distro package, as described in README.mpich-static.)
+
+Building outside of Cori for MPICH:
+If you have satisfied the prerequisites above, then do 'make -j mana'
+and continue as for Cori.
+
+===
+Testing and running is mostly the same as described at the top.
+However, if not running on a traditional HPC cluster,
+your site, you may need to replace 'srun' with 'mpirun'
+in the two commands for mana_launch and mana_restart, above.
+
+In a further complication, if using 'mpirun' with MPICH-derived MPI
+implementations, you may need to replace 'srun' with:
+  mpirun -iface ETHERNET_INTERFACE
+You should find the correct definition of MPIRUN in:
+  mpi-proxy-split/Makefile_config
+
+MANA tries to guess the required Ethernet interface in 'configure-mana'.
+Hopefully, that guess will be correct.  But see below for the issues, and
+under what circumstances MANA might guess the wrong interface and fail
+for an MPICH-derived MPI on a non-HPC cluster.
+[ This issue may now be solved now, using gethostbyname-static.tar.gz.
+  The issue used to arise because the lower half, lh_proxy, was compiled
+  statically with libc.a.  However, libc.a does not directly support
+  gethostbyname and getaddrinfo.  Instead, it tries to load libnss_*.so.
+  But we wanted lh_proxy to be statically linked so that it would consist
+  only of text/data/stack, and later calls to mmap. ]
+
+NOTE: On HPC clusters, it is more common to use 'srun', along with a custom
+      network interface (InfiniBand, Cray GNI, etc.).  So, most likely
+      Ethernet will not be used, and libc.a will not call libnss_*.so.
+
+The file MANA_ROOT/configure-mana tries to guess the Ethernet interface with:
```

Comment:
@fyshhh, Please check if feature/dmtcp-master still uses `configure-mana`.  Probably they don't use that file anymore and you would need to update this part, but I'm not sure.

Reply:
It hasn't been removed yet, so I'll leave it in.